lib/src/iterable_extensions.dart

```diff
@@ -601,6 +601,74 @@ extension IterableNumberExtension on Iterable<num> {
 ///
 /// Specialized version of some extensions of [IterableNumberExtension].
 extension IterableIntegerExtension on Iterable<int> {
+  /// A minimal element of the iterable, or `null` it the iterable is empty.
+  int? get minOrNull {
+    var iterator = this.iterator;
+    if (iterator.moveNext()) {
+      var value = iterator.current;
+      while (iterator.moveNext()) {
+        var newValue = iterator.current;
+        if (newValue < value) {
+          value = newValue;
+        }
+      }
+      return value;
+    }
+    return null;
+  }
+
+  /// A minimal element of the iterable.
+  ///
+  /// The iterable must not be empty.
+  int get min {
```

Comment:
Can be shortened to 
```dart
 int get min => minOrNull ?? throw StateError("No element");
```
Ditto for `max` below.

Reply:
Tried this but it complains that it expects an identifier where throw is. But I updated it to delegate to minOrNull / maxOrNull now.
![expects_identifier](https://user-images.githubusercontent.com/15617545/132725476-20bd5719-03a2-4946-aebb-000ac2413c42.png)

Reply:
My bad, I keep forgetting the precedence of `throw`. It can be written as:
```dart
  => maxOrNull ?? (throw StateErrro('No element');
```
(The error message is confusing because other things than identifiers can also occur, it basically means that it expects an expression with a different precedence).